plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/PredicatePushdownController.java

```diff
@@ -30,9 +33,19 @@
         }
         return new DomainPushdownResult(domain, Domain.all(domain.getType()));
     };
-    PredicatePushdownController CASE_INSENSITIVE_PUSHDOWN = (session, domain) -> new DomainPushdownResult(
-            domain.simplify(getDomainCompactionThreshold(session)),
-            domain);
+
+    PredicatePushdownController CASE_INSENSITIVE_PUSHDOWN = (session, domain) -> {
+        checkArgument(
+                domain.getType() instanceof VarcharType || domain.getType() instanceof CharType,
+                "CASE_SENSITIVE_PUSHDOWN can be used only for chars and varchars");
+        if (domain.getValues().isDiscreteSet()) {
```

Comment:
`domain.isNullableDiscreteSet()`

Reply:
I think the only difference between `domain.isNullableDiscreteSet()` and `domain.getValues().isDiscreteSet()` is that the first returns true for `Domain.onlyNull` case, which _is_ safe to pushdown and also safe to pushdown completely. 

Anyway, it's better to handle this case explicitly:

```
if (domain.isOnlyNull()) {
    return FULL_PUSHDOWN.apply(session, domain);
}
```